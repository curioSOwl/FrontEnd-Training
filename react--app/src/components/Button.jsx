import React from "react";

const Button = (props) => {
  return (
    <button
      onClick={props.handleSubmit}
      type="submit"
      className={props.className}
      data-testid="button-test-id"
    >
      {props.ButtonText}
    </button>
  );
};

export default Button;
