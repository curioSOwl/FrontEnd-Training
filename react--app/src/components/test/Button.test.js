import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../Button";

describe("Check if button works properly", () => {
  test("check if button rendered correctly", () => {
    const { getByTestId } = render(<Button />);
    const element = getByTestId("button-test-id");
    console.log(element);
    expect(element).toBeTruthy();
  });
  test("check if text displayed properly", () => {
    const text = "create";
    const { getByText } = render(<Button text={text} />);
    getByText(text);
  });
  test("check if onClick is trigerred", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Button onClick={onClick} />);
    const element = getByTestId("button-test-id");

    fireEvent.click(element);

    expect(onClick).toHaveBeenCalled();
  });
  test("check if snapshot is matched properly", () => {
    const onClick = jest.fn();
    const text = "Click";

    const { asFragment } = render(<Button text={text} onClick={onClick} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
