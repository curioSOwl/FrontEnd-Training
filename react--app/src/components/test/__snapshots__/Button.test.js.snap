// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if button works properly check if snapshot is matched properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-test-id"
    type="submit"
  />
</DocumentFragment>
`;
